plugins/modules/ec2_spot_instance_info.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/python
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://wwww.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: ec2_spot_instance_info
+version_added: 2.0.0
+short_description: Gather information about ec2 spot instance requests
+description:
+  - Describes the specified Spot Instance requests.
+author:
+  - Mandar Vijay Kulkarni (@mandar242)
+options:
+  filters:
+    description:
+      - A list of filters to apply.
+      - Each list item is a dict consisting of a filter key and a filter value.
+      - See U(https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeSpotInstanceRequests.html) for possible filters.
+    type: list
+    elements: dict
+  dry_run:
+    description:
+      - A boolean to check if you have the required permissions for the action.
+      - Does not make the request.
+      - If required permissions are not present then returns an error response UnauthorizedOperation else DryRunOperation.
+    default: False
+    choices: [ True, False ]
+    type: bool
+  spot_instance_request_ids:
+    description:
+      - One or more Spot Instance request IDs.
+    type: list
+    elements: str
+  next_token:
```

Comment:
Instead of using next_token and max_results we should use a paginator on the `describe_spot_instance_requests()` call in the module.
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Paginator.DescribeSpotInstanceRequests